doc/src/guides/solving/solve-numerically.md

```diff
@@ -0,0 +1,209 @@
+-1.00000000000000
+```
+
+## Solve a system of equations numerically
+
+### Solve multidimensional functions
+
+To solve multidimensional functions, supply a tuple of
+- functions `(f1, f2)`
+- variables to solve for `(x1, x2)`
+- starting values `(-1, 1)`
+
+```py
+>>> from sympy import Symbol, nsolve
+>>> x1 = Symbol('x1')
+>>> x2 = Symbol('x2')
+>>> f1 = 3 * x1**2 - 2 * x2**2 - 1
+>>> f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+>>> print(nsolve((f1, f2), (x1, x2), (-1, 1)))
+Matrix([[-1.19287309935246], [1.27844411169911]])
+```
+
+### Use SciPy on a lambda function *what is the advantage of this method?*
+
+You can *description*
+
+```py
+>>> from sympy.abc import x
+>>> from sympy.utilities.lambdify import implemented_function
+>>> from sympy import lambdify
+>>> from scipy import optimize
+>>> f = implemented_function('f', lambda x: x**3 - 1)
+>>> lam_f = lambdify(x, f(x))
+>>> sol = optimize.root_scalar(lam_f, bracket=[0, 2], method='brentq')
+>>> sol.root
+1.0
+```
+
+## Use the solution result
+
+### Substitute the result into an expression
+
+You can use [`subs`](sympy.core.basic.Basic.subs) to substitute numerical values
+into expressions to numerically evaluate them fully:
+
+```py
+>>> from sympy import cos, nsolve, Symbol, diff
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> f.subs(x, x_value) # Verify that root is correct
+0
+>>> derivative = diff(f, x)
+>>> derivative.subs(x, x_value) # Calculate the derivative where the function value is zero
+-1.67361202918321
+```
+
+or leave some symbols as variables:
+
+```py
+>>> from sympy import cos, nsolve, Symbol, diff
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> y = Symbol('y')
+>>> g = x * y**2
+>>> g.subs(x, x_value)
+0.739085133215161*y**2
+>>> derivative = diff(g, y)
+>>> derivative.subs(x, x_value)
+1.47817026643032*y
+```
+
+## Tradeoffs
+
+### Do not use `verify` for functions which are very steep near the root
+
+For functions which are very steep near the root, the verification of the
+solution may fail. In this case you should use the flag `verify=False` and
+independently verify the solution.
+
+```py
+>>> from sympy import cos, cosh, nsolve, Symbol
+>>> x = Symbol('x')
+>>> f = cos(x)*cosh(x) - 1
+>>> nsolve(f, 3.14*100)
+Traceback (most recent call last):
+...
+ValueError: Could not find root within given tolerance. (1.39267e+230 > 2.1684e-19)
+>>> ans = nsolve(f, 3.14*100, verify=False); ans
+312.588469032184
+>>> f.subs(x, ans).n(2)
+2.1e+121
+>>> (f/f.diff(x)).subs(x, ans).n(2)
+7.4e-15
+```
+
+## Not all equations can be solved
+
+{func}`~.nsolve` is a numerical solving function, so it is often the solution to
+equations which cannot be solved algebraically.
+
+### Equations with no solution
+
+Some equations have no solution, in which case SymPy may return an empty set.
+For example, the equation $x - 7 - x - 2 = 0$ reduces to $-9 = 0$, which has no
```

Comment:
In this case the equation trivially simplified to a number and `nsolve` doesn't accept that. But `sqrt(x) + 2` might be a better example since there is no solution for that.

Reply:
Thanks for the suggestion. Please help me understand--while both SymPy and [Wolfram Alpha](https://www.wolframalpha.com/input?i2d=true&i=Sqrt%5Bx%5D%2B2%3D0) agree with you, I thought the square root of 4 could be either 2 or -2, and if was chosen as -2, then 4 would be a solution of the equation you give?

Reply:
SymPy assumes that sqrt is only the principal square root. 

A more straightforward example might be `exp(x) = 0`, which doesn't have any (finite) solutions, even in a complex setting.

Reply:
Thanks, I went with `exp(x) = 0` for conceptual simplicity. A wrinkle is that it requires greater than default precision to return the correct response that there are no solutions. That led me to introduce `prec` earlier in the page, which I think could be valuable to many visitors.